### PR TITLE
Fix unbalanced braces, normalize many titles and abstracts, and add support for more TeX-math functions

### PR DESCRIPTION
**Pull request by mbollmann**
1. Finds and fixes a lot of instances of unbalanced braces, and **introduces a check for unbalanced braces during BibTeX generation**.
2. Runs normalization on titles and lots of newer abstracts, mostly to convert or remove TeX commands.
3. Adds some functionality to and refactors parts of TeX math handling.

- - - 
_Was originally:_

Handle `\mathrm`, `\textsc`, `\texttt`. Also, just drop the zero-argument variants `\bf`, `\rm` etc., as I don't see an easy way to support them, but they're probably not useful to display verbatim on the website.

### CODE REVIEW

File: tests/test_tex_math.py [hunk 1]
```diff
@@ -214,6 +214,14 @@
         '<tex-math>RoBERTa_{large}</tex-math>',
         '<span class="tex-math">RoBERTa<sub>large</sub></span>',
     ),
+    (
+        '<tex-math>RoBERTa_{\\rm large}</tex-math>',
+        '<span class="tex-math">RoBERTa<sub> large</sub></span>',
```
Comment (mbollmann) [inline tests/test_tex_math.py:219]: I would expect this extra space to be removed by `TexSoup`, but alas...

**Review by mjpost — APPROVED** [review]
LGTM. Would a fix to #2505 make sense within this PR, too?

**Comment by mbollmann** [comment]
Yeah, possibly. I can look into that one.

**Comment by mbollmann** [comment]
Added support for more TeX functions and refactored the code to make it a bit more readable.

I also fixed all the instances of mismatched braces I could find; I'll check what kind of unsupported TeX commands are left in the data after the preview finishes building.

:warning: **I also added a check for unbalanced braces during BibTeX generation.** This could potentially make **future ingestions fail** when they have this issue. The alternative would be just generating a warning and leaving out the affected abstracts during the BibTeX export, though this might confuse people when they find an abstract missing in our data.

**Comment by mbollmann** [comment]
Oof, I completely ignored that TeX functions are only supported inside `<tex-math>` expressions anyway, so the fixes don't actually improve much ... putting this on hold for a little bit longer.

**Comment by mbollmann** [comment]
I ran David's helpful normalization script on all titles and many abstracts (mostly the newer volumes, as that's where the unbalanced braces were), converting a lot of TeX functions in the process.

I spot-checked the results and manually reverted a few wrong modifications, but there may be others. Some common pitfalls:

1. Markup like `\texttt` or `\textsc` that we don't support outside of math expressions is just removed.
2. Some curly braces that are actually meaningful (meant as verbatim curly braces in the text, e.g. "translating de-{en, fr}") are removed.

Overall the benefits outweigh the drawbacks here, I think, and this could potentially be merged now. @mjpost

**Pull request merged**